### DISCUSSION TITLE
tailwind 알아보기

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#92] [프론트 기술 질문 공간]
tailwind를 왜 많은 개발자들이 선호하고 사용할까요?
장점과 단점을 비교하고 개인의 의견도 덧붙여주세요~

**Comment by michoball** [comment]
tailwind는 utility class 로 css 스타일링을 할 수 있게 해줘 개발자가 스타일링에 쏟는 시간을 대폭 절약해줍니다. 
build를 할 때에도 실제 사용되는 스타일만 build.css 파일로 만들어져서 build된 파일의 크기를 줄여줍니다. 

모던 프론트엔드 프레임워크 & 라이브러리를 사용하면 똑같은 스타일링 코드를 반복해야한다는 단점도 보완이 되고, 
tailwind.config.js 에서 커스터마이징도 할 수가 있어서 좀 더 깔끔하게 사용이 가능합니다. 

개인적으로 class 안에 너무 많은 텍스트가 들어가 코드의 가독성을 헤친다는 점에서 큰 단점이라고 생각하고 있지만 많은 부분 커스터마이징을 통해 개선될 수 있다는 점에서 또 기본적으로  css 코드 작성의 시간을 대폭 줄여준다는 점에서 tailwind를 사용하는 것이 좋다고 생각합니다.

**Comment by WooGie911** [comment]
Tailwind CSS는 Utility-First 컨셉으로 스타일 코드도 HTML 코드 안에 있기 때문에 HTML와 CSS 파일을 별도로 관리할 필요가 없습니다. 
덕분에 HTML-CSS를 왔다갔다하며 시간을 쓰거나 위의 이미지처럼 화면을 분할해서 사용하지 않아도 됩니다. 

또한 원하는 태그의 스타일을 변경하기 위해 클래스명을 검색해가며 일일이 필요한 CSS 코드를 찾을 수고도 사라집니다.

게다가 Tailwind CSS를 사용하면 랩핑 태그의 클래스명을 사용할 일이 거의 없으므로 `container`, `wrapper`, `inner-wrapper`와 같은 클래스명을 고민하지 않아도 된다는 장점이 있습니다.

뿐만 아니라 Tailwind CSS는 다른 프레임워크들에 비해 기본 스타일 값을 디테일한 부분까지 쉽게 커스텀이 가능합니다. 
또한 커스텀을 할 때 기본 스타일 값을 수정하는 방식이기에 디자인 일관성도 해치지 않습니다. 


다만 부분적으로 반복되는 스타일링 코드를 반복해야 한다는 점과 결국 가독성이 떨어지고 코드가 지저분해 보인다는 단점이 있습니다.

뿐만 아니라 새로운 스타일을 사용하면 CSS 클래스를 생성하게 되어 CSS 파일 크기가 커지고 페이지 로드 시간도 늘어날 수 있으며, 다른 개발자들과 협업할 경우 CSS파일에 충돌이 생길 수 있습니다.

또한 Tailwind CSS를 사용할 수 없는 스타일인 경우에 별도의 CSS 파일을 생성해야 하기 때문에 일관성 있는 스타일링이 어려울 수도 있습니다.


이와 같이 장단점이 확실한 라이브러리이지만 사용해보면 작업속도가 확실히 빠르기 때문에 가독성이 떨어지더라도 상당히 매력적인 라이브러리라고 생각됩니다.
